assets/js/persistence.js

```diff
@@ -171,6 +180,10 @@ function restoreChoices(mode) {
             $('#originalText').val(decodeURIComponent(getURLParam('q')));
         }
 
+        if(getURLParam('qP').length > 0) {
```

Comment:
Um, should I create a separate function with modes as argument for this? - using the lookup table here will make a lot of code repetitive? Is this what you are saying?

For eg:

```
if(getURLParam(('q' + hashURLMap[hash]).length > 0) {

}
```